dispatcher/src/main/java/io/zeebe/dispatcher/Dispatcher.java

```diff
@@ -233,45 +211,34 @@ public long claim(ClaimedFragment claim, int length) {
    *     return value is negative.
    */
   public long claim(ClaimedFragment claim, int length, int streamId) {
-    final long limit = publisherLimit.get();
```

Comment:
So the main difference I see is that `claim` was not previously checking if the dispatcher was closed, but now it will. I'm not super familiar with the dispatcher, could this introduce new unexpected behaviour? I mean I don't see how you could claim things when it's closed or why you would :sweat_smile:

Reply:
yes and no. Normal claim had no check the claim for batches had one, so I think it was a bug.